### PR TITLE
Fix batch operations when advanced visibility is not used

### PR DESCRIPTION
**Pull request by feedmeapples**
## What was changed

Removed reliance on `CountWorkflow` API when starting batch jobs

## Why?

Fixes batch jobs for cases when ES is disabled (including dev server)

## Checklist

1. Closes 

2. How was this tested:

manually. requires dev server e2e setup for proper tests 

3. Any docs updates needed?

### CODE REVIEW

File: batch/batch_commands.go [hunk 1]
```diff
@@ -151,14 +151,14 @@ func startBatchJob(c *cli.Context, req *workflowservice.StartBatchOperationReque
...
 		Query:     query,
 	})
-	if err != nil {
-		return fmt.Errorf("unable to count impacted workflows: %w", err)
+
+	var promptMsg string
+	if err == nil {
+		promptMsg = fmt.Sprintf("Will start a batch job operating on %v Workflow Executions. Continue? Y/N", color.Yellow(c, "%v", count.GetCount()))
+	} else {
+		promptMsg = fmt.Sprintf("Will start a batch job on all Workflow Executions that match the %s. Continue? Y/N", color.Yellow(c, "%v", query))
```
Comment (bergundy) [inline batch/batch_commands.go:159]: Hmm.. I didn't know we were prompting here.
Is there a way to disable the prompt?

**Comment by bergundy** [comment]
Can we add a test for this please?

File: batch/batch_commands.go (hunk 1 above)
Comment (feedmeapples) [inline batch/batch_commands.go:159]:
```suggestion
		promptMsg = fmt.Sprintf("Will start a batch job on all Workflow Executions that match %s. Continue? Y/N", color.Yellow(c, "%v", query))
```

Comment (feedmeapples) [reply batch/batch_commands.go]: it can be disabled if the `--yes` flag is passed

File: batch/batch_commands.go [hunk 2]
```diff
@@ -151,14 +151,14 @@ func startBatchJob(c *cli.Context, req *workflowservice.StartBatchOperationReque
...
 		Query:     query,
 	})
-	if err != nil {
-		return fmt.Errorf("unable to count impacted workflows: %w", err)
+
+	var promptMsg string
+	if err == nil {
+		promptMsg = fmt.Sprintf("Will start a batch job operating on %v Workflow Executions. Continue? Y/N", color.Yellow(c, "%v", count.GetCount()))
+	} else {
+		promptMsg = fmt.Sprintf("Will start a batch job on all Workflow Executions that match %s. Continue? Y/N", color.Yellow(c, "%v", query))
```
Comment (alexshtin) [inline batch/batch_commands.go:159]:
```suggestion
		promptMsg = fmt.Sprintf("Unable to count impacted workflows: %s. Will start a batch job on all Workflow Executions that match %s. Continue? Y/N", err, color.Yellow(c, "%v", query))
```

**Review by alexshtin — APPROVED** [review]

**Comment by alexshtin** [comment]
Just FYI, @rodrigozhou added support for `Count` to SQL based advanced visibility. Standard visibility will be deprecated soon and `Count` should be always supported.

**Pull request merged**